src/App.tsx

```diff
@@ -1,12 +1,79 @@
 import React from 'react';
 import './App.scss';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './react-app-env';
+import GoodList from './components/GoodList/GoodList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+interface State {
+  isStarted: boolean,
+  goods: Good[] | [],
+}
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component<{}, State> {
+  state: State = {
+    isStarted: false,
+    goods: [],
+  };
+
+  updateState = (callback: () => Promise<Good[] | []>) => {
+    callback().then((goods: Good[]) => {
```

Comment:
unless a library or legacy codebase forces you to use then/catch, the better choice for readability and maintainability is async/await